src/impl/tpls/KokkosBlas_geqrf_tpl_spec_avail.hpp

```diff
@@ -0,0 +1,164 @@
+KOKKOSBLAS_GEQRF_WORKSPACE_TPL_SPEC_AVAIL_LAPACK(float, Kokkos::LayoutRight,
+                                                 Kokkos::HostSpace)
+KOKKOSBLAS_GEQRF_TPL_SPEC_AVAIL_LAPACK(float, Kokkos::LayoutRight,
+                                       Kokkos::HostSpace)
+#endif
+
+#if defined(KOKKOSKERNELS_INST_KOKKOS_COMPLEX_DOUBLE_) && \
+    defined(KOKKOSKERNELS_INST_LAYOUTRIGHT)
+KOKKOSBLAS_GEQRF_WORKSPACE_TPL_SPEC_AVAIL_LAPACK(Kokkos::complex<double>,
+                                                 Kokkos::LayoutRight,
+                                                 Kokkos::HostSpace)
+KOKKOSBLAS_GEQRF_TPL_SPEC_AVAIL_LAPACK(Kokkos::complex<double>,
+                                       Kokkos::LayoutRight, Kokkos::HostSpace)
+#endif
+
+#if defined(KOKKOSKERNELS_INST_KOKKOS_COMPLEX_FLOAT_) && \
+    defined(KOKKOSKERNELS_INST_LAYOUTRIGHT)
+KOKKOSBLAS_GEQRF_WORKSPACE_TPL_SPEC_AVAIL_LAPACK(Kokkos::complex<float>,
+                                                 Kokkos::LayoutRight,
+                                                 Kokkos::HostSpace)
+KOKKOSBLAS_GEQRF_TPL_SPEC_AVAIL_LAPACK(Kokkos::complex<float>,
+                                       Kokkos::LayoutRight, Kokkos::HostSpace)
+#endif
+
+#endif  // if BLAS && LAPACK
+
+// CUSOLVER
+//
+#if defined(KOKKOSKERNELS_ENABLE_TPL_CUSOLVER)
+
+#define KOKKOSBLAS_GEQRF_TPL_SPEC_AVAIL_CUSOLVER(SCALAR, LAYOUTA, MEMSPACE) \
+  template <class ExecSpace>                                                \
+  struct geqrf_tpl_spec_avail<                                              \
+      Kokkos::View<SCALAR**, LAYOUTA, Kokkos::Device<ExecSpace, MEMSPACE>,  \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,               \
+      Kokkos::View<SCALAR*, LAYOUTA, Kokkos::Device<ExecSpace, MEMSPACE>,   \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,               \
+      Kokkos::View<SCALAR*, LAYOUTA, Kokkos::Device<ExecSpace, MEMSPACE>,   \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> > > {            \
+    enum : bool { value = true };                                           \
+  };
+
+#define KOKKOSBLAS_GEQRF_WORKSPACE_TPL_SPEC_AVAIL_CUSOLVER(SCALAR, LAYOUTA, \
+                                                           MEMSPACE)        \
+  template <class ExecSpace>                                                \
+  struct geqrf_workspace_tpl_spec_avail<                                    \
+      Kokkos::View<SCALAR**, LAYOUTA, Kokkos::Device<ExecSpace, MEMSPACE>,  \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> >,               \
+      Kokkos::View<SCALAR*, LAYOUTA, Kokkos::Device<ExecSpace, MEMSPACE>,   \
+                   Kokkos::MemoryTraits<Kokkos::Unmanaged> > > {            \
+    enum : bool { value = true };                                           \
+  };
+
+KOKKOSBLAS_GEQRF_WORKSPACE_TPL_SPEC_AVAIL_CUSOLVER(double, Kokkos::LayoutLeft,
```

Comment:
I think these should be made available for `Kokkos::CudaUVMSpace`

Reply:
Whoops, I guess I thought CudaUVMSpace inherited from CudaSpace. Thanks I've added these now.